src/exo/LoopIR_scheduling.py

```diff
@@ -235,7 +235,28 @@ def _compose(f, g):
     return lambda x: f(g(x))
 
 
-def _replace_pats(ir, fwd, c, pat, repl, c_is_fwded=False):
+def _replace_helper(c, c_repl, attrs):
+    if attrs:
+        ir, fwd_s = c.get_root(), lambda x: x
+        for attr in attrs:
+            if attr in ["body", "orelse", "idx"]:
+                ir, fwd_attr = c._child_block(attr)._replace(c_repl[attr])
+            else:
+                ir, fwd_attr = c._child_node(attr)._replace(c_repl[attr])
+            fwd_s = _compose(fwd_attr, fwd_s)
+        return ir, fwd_s
+    else:
+        if (
+            isinstance(c, ic.Block)
+            and not isinstance(c_repl, list)
+            or isinstance(c, ic.Node)
+            and c.get_index() is not None
+        ):
+            c_repl = [c_repl]
+        return c._replace(c_repl)
```

Comment:
This seems like something that should ultimately be folded into the design of the internal cursors. But it's okay for now, since it appears to work.